### PR TITLE
NFDIV-693:

### PR DESCRIPTION
**Pull request by tomxelliott**
Use a mock payment to move the case to Submitted in the solicitor-statement-of-truth-pay-submit event

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-693

### CODE REVIEW

**Review by linusnorton — APPROVED** [review]

**Review by nitinprabhu — COMMENTED** [review]
can we have some unit test for `SolicitorSubmitApplicationService.java`?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitApplicationService.java [hunk 1]
```diff
@@ -83,9 +85,18 @@ public OrderSummary getOrderSummary() {
         applicantSubmittedNotification.send(caseData, caseId);
         solicitorSubmittedNotification.send(caseData, caseId);
 
+        State state = SolicitorAwaitingPaymentConfirmation;
+
+        List<String> submittedErrors = Submitted.validate(caseData);
+        if (submittedErrors.isEmpty()) {
+            caseData.setDateSubmitted(LocalDateTime.now());
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitApplicationService.java:92]: You could use an Autowired Clock, and use that to create the LocalDateTime, e.g. LocalDateTime.now(clock).  You can then unit test this easily.
See ClockConfiguration class for the default autowired clock instance.

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitApplicationService.java:92]: Will take a look, thanks

**Review by mapingo — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/common/model/State.java [hunk 2]
```diff
@@ -60,6 +60,12 @@
             if (hasAwaitingDocuments(caseData)) {
                 errors.add("Awaiting documents");
             }
+            if (!caseData.hasStatementOfTruth()) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/common/model/State.java:63]: surely these things are either or - in that either the citizen needs to have done the SOT or the solicitor needs to have. Probably not both

File: src/main/java/uk/gov/hmcts/divorce/common/validation/ValidationUtil.java [hunk 3]
```diff
@@ -132,6 +132,7 @@ public static boolean isPaymentIncomplete(CaseData caseData) {
     }
 
     public static boolean hasAwaitingDocuments(CaseData caseData) {
+        // TODO - use .equals() for string comparison instead of ==
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/common/validation/ValidationUtil.java:135]: I can't remember whether we did but I think we were going to do a .toBool on YesOrNo that would be easier

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/common/validation/ValidationUtil.java:135]: Used the .toBoolean() method in the YesOrNo model 👍

File: src/main/java/uk/gov/hmcts/divorce/common/model/State.java (hunk 2 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/common/model/State.java:63]: My misunderstanding - have updated

**Review by linusnorton — APPROVED** [review]

**Review by nitinprabhu — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/divorce/citizen/event/PaymentMadeTest.java [hunk 4]
```diff
@@ -81,6 +81,7 @@
     public void givenValidCaseDataWhenCallbackIsInvokedThenSendEmail() throws Exception {
         Map<String, Object> data = caseDataMap();
         data.put("dateSubmitted", LocalDateTime.now());
+        data.put("solSignStatementOfTruth", YesOrNo.YES);
```
Comment (mapingo) [inline src/integrationTest/java/uk/gov/hmcts/divorce/citizen/event/PaymentMadeTest.java:84]: YesOrNo could be static imported

**Review by linusnorton — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/divorce/citizen/event/PaymentMadeTest.java (hunk 4 above)
Comment (tomxelliott) [reply src/integrationTest/java/uk/gov/hmcts/divorce/citizen/event/PaymentMadeTest.java]: Have updated 👍

**Pull request merged**